src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1833,10 +1836,40 @@ internal ImmutableArray<BoundNode> GetBoundNodes(CSharpSyntaxNode node)
             var statementBinder = GetEnclosingBinder(GetAdjustedNodePosition(nodeToBind));
             Binder incrementalBinder = new IncrementalBinder(this, statementBinder);
 
+            // PROTOTYPE(nullable-api): Speculative models are going to have to do something more advanced
+            // here. They will need to run nullable analysis up to the point that is being speculated on, and
+            // then take that state and run analysis on the statement or expression being speculated on.
+            // Currently, it will return incorrect info because it's just running analysis on the speculated
+            // part.
+
+            BoundNode root;
             using (_nodeMapLock.DisposableWrite())
             {
-                BoundNode boundStatement = this.Bind(incrementalBinder, nodeToBind, _ignoredDiagnostics);
-                results = GuardedAddBoundTreeAndGetBoundNodeFromMap(node, boundStatement);
+                root = this.Bind(incrementalBinder, nodeToBind, _ignoredDiagnostics);
+            }
+
+            if (EnableNullableAnalysis)
+            {
+                if (MemberSymbol is MethodSymbol method)
+                {
+                    BoundNode rewritten = NullableWalker.AnalyzeAndRewrite(Compilation, method, root, RootBinder.Conversions, new DiagnosticBag());
+                    using (_nodeMapLock.DisposableWrite())
+                    {
+                        results = GuardedAddBoundTreeAndGetBoundNodeFromMap(node, rewritten);
+                    }
+                }
+                else
+                {
+                    // PROTOTYPE(nullable-api): Handle other symbol types
+                    using (_nodeMapLock.DisposableWrite())
+                    {
+                        _ = GuardedAddBoundTreeAndGetBoundNodeFromMap(node, root);
+                    }
```

Comment:
Please consider moving this after the `if`, to share the update across both branches.